backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/RunVmCommand.java

```diff
@@ -956,10 +958,20 @@ protected boolean getVdsToRunOn() {
         }
 
         addCpuAndNumaPinning();
+        setDedicatedCpus();
 
         return true;
     }
 
+    private void setDedicatedCpus() {
+        if (getVm().getCpuPinningPolicy() != CpuPinningPolicy.DEDICATED) {
+            return;
+        }
+        List<VdsCpuUnit> vdsCpuUnits = getVdsManager().getCpuTopology().stream()
+                .filter(cpu -> getVmId().equals(cpu.getVmId())).sorted().collect(Collectors.toList());
```

Comment:
The cpu.getVmiId() returns the list of vm ids - shouldn't it be `filter(cpu -> cpu.getVmId().contains(getVmId()))`